packages/core/src/types.ts

```diff
@@ -1349,3 +1350,29 @@ export interface Behavior<TEvent extends EventObject, TEmitted = any> {
   initialState: TEmitted;
   start?: (actorCtx: ActorContext<TEvent, TEmitted>) => TEmitted;
 }
+
+export type EventsFrom<T> = EventObject &
+  (T extends StateMachine<any, any, infer TEvents0, any>
+    ? TEvents0
+    : T extends Model<any, infer TEvents1, any>
```

Comment:
Within all of those types here, we use `TEvent` (singular) and not `TEvents`, I would probably go for consistency and rename all those occurrences here to a singular form. This also brings up a question - maybe this helper should be named `EventFrom` and not `EventsFrom`? 🤔

Reply:
e308b9b